platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java

```diff
@@ -380,8 +388,50 @@ public static boolean hasAnyCauseSuppliedType(
                 .map(address -> new PeerInfo(
                         address.getNodeId(),
                         address.getSelfName(),
-                        address.getHostnameExternal(),
-                        address.getSigCert()))
+                        Objects.requireNonNull(address.getHostnameExternal()),
+                        Objects.requireNonNull(address.getSigCert())))
                 .toList();
     }
+
+    /**
+     * Validates that the client on the other end of the socket possesses an agreement
+     *  certificate, and that the agreement certificate's issuer's principal matches
+     *  the signing certificate's subject for one or more known peers
+     *
+     * @param sslSocket
```

Comment:
Javadocs for these parameters needs some improvement. The description for `sslSocket` is incorrect, and the description for `peerInfoList` is misleading. The peers in this list are the valid peers of the network that the client on the other side of the socket must be one of. The method does not actually validate the list of peers as the comment states. Finally, the return value could be `null`. The javadoc for the return value should document what a `null` return value means.